# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1701013259 -3600
# Parent  03ee6c1eaa13b4965f2b43502029e35a0ace6730
Misc: Use Betterbird's own donation appeal URL.
* * *
Misc: Use own prefs for BB donation appeal.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1395,17 +1395,21 @@ pref("services.sync.engine.calendars", t
 pref("services.sync.engine.creditcards", false);
 pref("services.sync.engine.identities", true);
 pref("services.sync.engine.prefs", false);
 #endif
 
 // Donation appeal.
 pref("app.donation.eoy.version", 3);
 pref("app.donation.eoy.version.viewed", 0);
-pref("app.donation.eoy.url", "https://www.thunderbird.net/thunderbird/115.0/eoy/");
+// Use our own prefs for the end-of-year appeal. Otherwise running TB on the
+// profile would show the TB appeal and then never again the BB appeal.
+pref("betterbird.donation.eoy.version", 2022);
+pref("betterbird.donation.eoy.version.viewed", 0);
+pref("betterbird.donation.eoy.url", "https://www.betterbird.eu/end-of-year/");
 
 // IMAP-JS disabled, Bug 1707547.
 pref("mailnews.imap.jsmodule", false);
 
 // Unified toolbar
 
 // 0: icons beside text
 // 1: icons above text
diff --git a/mail/base/content/messenger.js b/mail/base/content/messenger.js
--- a/mail/base/content/messenger.js
+++ b/mail/base/content/messenger.js
@@ -441,46 +441,64 @@ var gMailInit = {
   },
 
   /**
    * Check if we can trigger the opening of the donation appeal page.
    *
    * @returns {boolean} - True if the donation appeal page should be opened.
    */
   shouldShowEOYDonationAppeal() {
-    let currentEOY = Services.prefs.getIntPref("app.donation.eoy.version", 1);
+    // Suppress TB appeal.
+    let currentEOY = Services.prefs.getIntPref("app.donation.eoy.version", 999);
+    Services.prefs.setIntPref("app.donation.eoy.version.viewed", currentEOY);
+
+    currentEOY = Services.prefs.getIntPref(
+      "betterbird.donation.eoy.version",
+      2022
+    );
     let viewedEOY = Services.prefs.getIntPref(
-      "app.donation.eoy.version.viewed",
+      "betterbird.donation.eoy.version.viewed",
       0
     );
 
-    // True if the user never saw the donation appeal, this is not a new
-    // profile (since users are already prompted to donate after downloading),
-    // and we're not running tests.
-    return (
-      viewedEOY < currentEOY &&
-      !specialTabs.shouldShowPolicyNotification() &&
-      !Cu.isInAutomation
+    // Short-circuit further checking.
+    if (viewedEOY >= currentEOY) {
+      return false;
+    }
+
+    let now = Math.round(Date.now() / 1000);
+    // If we never checked, pretend we checked now.
+    let lastCheckTime = Services.prefs.getIntPref(
+      "betterbird.update.lastCheckTime",
+      now
     );
+
+    // Display one day after installation the earliest. 24 * 60 * 60.
+    return now - lastCheckTime > 86400 && !Cu.isInAutomation;
   },
 
   /**
    * Open the end of year appeal in a new web browser page. We don't open this
    * in a tab due to the complexity of the donation site, and we don't want to
    * handle that inside Thunderbird.
    */
   showEOYDonationAppeal() {
-    let url = Services.prefs.getStringPref("app.donation.eoy.url");
+    let url = Services.prefs.getStringPref("betterbird.donation.eoy.url");
     let protocolSvc = Cc[
       "@mozilla.org/uriloader/external-protocol-service;1"
     ].getService(Ci.nsIExternalProtocolService);
     protocolSvc.loadURI(Services.io.newURI(url));
-
-    let currentEOY = Services.prefs.getIntPref("app.donation.eoy.version", 1);
-    Services.prefs.setIntPref("app.donation.eoy.version.viewed", currentEOY);
+    let currentEOY = Services.prefs.getIntPref(
+      "betterbird.donation.eoy.version",
+      2022
+    );
+    Services.prefs.setIntPref(
+      "betterbird.donation.eoy.version.viewed",
+      currentEOY
+    );
   },
 };
 
 /**
  * Called at startup to verify if we have ny existing account, even if invalid,
  * and if not, it will trigger the Account Hub in a tab.
  *
  * @returns {boolean} - True if we have at least one existing account.
